Netkan/Processors/QueueHandler.cs

```diff
@@ -0,0 +1,197 @@
+            {
+                log.DebugFormat("Metadata returned: {0}", msg.Body);
+                var netkan = new Metadata(JObject.Parse(msg.Body));
+
+                int releases = 1;
+                MessageAttributeValue releasesAttr;
+                if (msg.MessageAttributes.TryGetValue("Releases", out releasesAttr))
+                {
+                    releases = int.Parse(releasesAttr.StringValue);
+                }
+
+                log.InfoFormat("Inflating {0}", netkan.Identifier);
+                IEnumerable<Metadata> ckans = null;
+                try
+                {
+                    ckans = inflator.Inflate($"{netkan.Identifier}.netkan", netkan, releases);
+                }
+                catch (Exception e)
+                {
+                    e = e.GetBaseException() ?? e;
+                    log.InfoFormat("Inflation failed, sending error: {0}", e.Message);
+                    sendCkan(null, netkan, false, e.Message);
+                }
+                if (ckans != null)
+                {
+                    foreach (Metadata ckan in ckans)
+                    {
+                        log.InfoFormat("Sending {0}-{1}", ckan.Identifier, ckan.Version);
+                        sendCkan(ckan, netkan, true);
+                    }
+                }
+
+                // Delete message after processing
+                log.Debug("Deleting message");
+                client.DeleteMessage(new DeleteMessageRequest()
+                {
+                    QueueUrl      = url,
+                    ReceiptHandle = msg.ReceiptHandle,
+                });
+            }
+        }
+
+        private void sendCkan(Metadata ckan, Metadata netkan, bool success, string err = null)
+        {
+            var attribs = new Dictionary<string, MessageAttributeValue>()
+            {
+                {
+                    "ModIdentifier",
+                    new MessageAttributeValue()
+                    {
+                        DataType    = "String",
+                        StringValue = netkan.Identifier
+                    }
+                },
+                {
+                    "Staged",
+                    new MessageAttributeValue()
+                    {
+                        DataType    = "String",
+                        StringValue = netkan.Staged.ToString()
+                    }
+                },
+                {
+                    "Success",
+                    new MessageAttributeValue()
+                    {
+                        DataType    = "String",
+                        StringValue = success.ToString()
+                    }
+                },
+                {
+                    "CheckTime",
+                    new MessageAttributeValue()
+                    {
+                        DataType    = "String",
+                        StringValue = DateTime.UtcNow.ToString("s", CultureInfo.InvariantCulture)
+                    }
```

Comment:
As an extra attribute, I think what would really be good to have is the resulting ckan filename. Currently I do some funky git gyrations to figure it out, which really isn't necessary if NetKAN can tell me.

That way the indexer is always going to do exactly what NetKAN would do.

Reply:
We can do that. With or without the output dir parameter?

Reply:
Added FileName attribute, which will include the output dir parameter if any. Let me know if you want it changed to exclude the output dir.